tcms/rpc/api/testexecution.py

```diff
@@ -183,12 +183,25 @@ def update(execution_id, values, **kwargs):
 
     # if this call updated TE.status then adjust timestamps
     if values.get("status"):
+        now = timezone.now()
         if test_execution.status.weight != 0:
-            test_execution.close_date = timezone.now()
+            test_execution.close_date = now
         else:
             test_execution.close_date = None
         test_execution.save()
 
+        all_runs = TestExecution.objects.filter(run=test_execution.run)
+        if test_execution.status.weight > 0 or test_execution.status.weight < 0:
```

Comment:
why not .weight != 0  ?

Reply:
The above should be `test_execution.status.weight != 0 and not all_executions.filter(status_weight=0).exists()` and avoid the nested if condition.